indico/filters/__init__.py

```diff
@@ -0,0 +1,51 @@
+from enum import Enum
+from typing import Any, Dict, Iterable
+
+
+class FilterMode(Enum):
+    AND = "and"
+    OR = "or"
+
+
+class BaseFilter(dict):
```

Comment:
Mmm so this a bit different then I expected in terms of interface, I was thinking the filters would be fields on a Filter object so like SubmissionFilter as a class with known filterable and safe fields.

I think the interface I was thinking is like

SubmissionFilter(inputFilename=["test.pdf"], status=["complete"])

or 

filter = SubmissionFilter()
filter.status = ["complete"] and if status has type annotation it would know how to figure it out.